src/Orleans.Core/Serialization/BinaryTokenStreamWriter2.cs

```diff
@@ -250,32 +250,30 @@ public void WriteTypeHeader(Type t, Type expected = null)
             this.Write(typeKey.Length);
             this.Write(typeKey);
         }
-                
+
         public void Write(byte[] b, int offset, int count)
         {
             if (count <= 0)
             {
                 return;
             }
 
-            if ((offset == 0) && (count == b.Length))
+            if (offset == 0 && count == b.Length)
             {
                 this.Write(b);
             }
             else
             {
-                var temp = new byte[count];
-                Buffer.BlockCopy(b, offset, temp, 0, count);
-                this.Write(temp);
+                this.Write(new ArraySegment<byte>(b, offset, count));
```

Comment:
Any reason to create an ArraySegment<byte>, and immediately let it implicitly convert, instead of directly constructing the Span?

```suggestion
                this.Write(new ReadOnlySpan<byte>(b, offset, count));
```